### PR TITLE
Feat/support role based authentication

### PR DESCRIPTION
**Pull request by Credgate**
Added role based support for all AWS S3 interactions.

You can now pass in the following arguments `role_arn` and `intermediary_role`. This will override any `aws_access_key_id`, `aws_secret_access_key`, and `aws_session_token`  you pass in. The reason for this is because the role that you assume is only valid for ~1 hour. If the keys expire then the `target-redshift` operation would need to get them again. It can do that using the roles to obtain the keys.

assume intermediary_role -> assume external role -> create/update S3 client -> Interact with AWS resources.

I've tested this with the new redshift connector and it works as intended.
I've also tested an existing input connector and confirmed that the existing implementation also works without passing the two new arguments.

### CODE REVIEW

File: target_redshift/db_sync.py [hunk 1]
```diff
@@ -688,20 +710,20 @@ def update_columns(self):
             self.table_cache = self.get_table_columns(filter_schemas=[self.schema_name])
 
     def drop_column(self, column_name, stream):
-        drop_column = "ALTER TABLE '{}' DROP COLUMN {}".format(self.table_name(stream, is_stage=False), column_name)
+        drop_column = "ALTER TABLE {} DROP COLUMN {}".format(self.table_name(stream, is_stage=False), column_name)
```
Comment (shanemorton-au) [inline target_redshift/db_sync.py:713]: Why remove the quotes? Will this still work if the table name has reserved chars in it?

File: target_redshift/db_sync.py [hunk 2]
```diff
@@ -738,3 +760,48 @@ def sync_table(self):
         else:
             self.logger.info("Table '{}' exists".format(self.schema_name))
             self.update_columns()
+
+    def check_and_return_credentials(self, settings) -> dict[str, Any]:
```
Comment (shanemorton-au) [inline target_redshift/db_sync.py:764]: What if the target redshift cluster is in a different region? Does this still work?

**Review by shanemorton-au — DISMISSED** [review]
A few questions but LGTM otherwise

File: target_redshift/db_sync.py (hunk 2 above)
Comment (Credgate) [reply target_redshift/db_sync.py:764]: We interact with redshift via connection through pyscopg2 it does not use Boto3

STS is global service so i don't believe it matters what region_name we choose for the client but from past experience it was a required argument.

File: target_redshift/db_sync.py (hunk 1 above)
Comment (Credgate) [reply target_redshift/db_sync.py]: 🤔 Ahh I think my IDE linter did this on me. It looks like it works fine without but to be safe I put them back.

**Review by santiagocasti — APPROVED** [review]
Please do extensive testing in main with all connectors using regular credentials.

**Pull request merged**